### DISCUSSION TITLE
Crowdsec, bouncer only

### DISCUSSION THREAD

**Discussion by danielholm** [ZoeyVid/NPMplus#1072] [Q&A] [answered]
Hi and thank you for a very nice fork. After migration it seem to work flawlessly.

I was wondering if a short guide of setting up a Crowdsec bouncer only, could be added?
I got it setup after a bit of work - in my case I had different path's than in the guide. But is it needed to add the crowdsec container or is the bouncer already part of the npmplus image?

**Comment by Zoey2936** [comment, marked as answer]
Hi,
crowdsec has two parts:
itself and bouncers.
NPMplus itself only contains the bouncer.
Bouncers only ask the the local ip of crowdsec itself if an external ip should be blocked or not and if yes, it blocks it
Crowdsec itself provides this local api and parses logs to see which ips are bad and should be blocked.
The Readme provides a compose which sets up crowdsec itself with configuration to parse npmplus logs

**Reply by danielholm** [reply]
I got Crowdsec with LAPI installed on my pfsense firewall. So I figured I just needed the bouncer in this container, where the logs are processed elsewhere. I did follow the readme and added the crowdsec image as well. But thanks to your comment, I am now thinking about changing things. It did however provided me with the answer regarding if a bouncer is in the image, which indeed it is.

**Reply by Zoey2936** [reply]
If you alread, have a local api of crowdsec, yes, then you don't need a second one, just make sure to adapt the configuration so crowdsec can read npmplus logs and npmplus can access the local api

**Reply by danielholm** [reply]
I am trying, but the config is overwritten on each restart of crowdsec. Is there some env variable that I need to set?

**Reply by Zoey2936** [reply]
why config exactly?

**Reply by danielholm** [reply]
What do you mean? The settings for crowdsec are reset while restart.

**Reply by Zoey2936** [reply]
I mean "which config exectly"?

**Comment by danielholm** [comment]
Is this, inside the containter, the correct path to bouncer-config?
>> /data/crowdsec/crowdsec-openresty-bouncer.conf
